llvm/lib/Target/X86/X86Subtarget.cpp

```diff
@@ -252,7 +252,7 @@ void X86Subtarget::initSubtargetFeatures(StringRef CPU, StringRef TuneCPU,
     CPU = "generic";
 
   if (TuneCPU.empty())
-    TuneCPU = "i586"; // FIXME: "generic" is more modern than llc tests expect.
+    TuneCPU = Is64Bit ? "x86-64" : "i586";
```

Comment:
64-bit should use "generic" tuning.

Reply:
Done!

Reply:
Did you check the lit tests? I found some 64-bit test changes in your previous patch. So I assume they are affected by this change too.